Lib/test/test_dataclasses/__init__.py

```diff
@@ -1524,6 +1524,24 @@ class A(types.GenericAlias):
         self.assertTrue(is_dataclass(type(a)))
         self.assertTrue(is_dataclass(a))
 
+    def test_is_dataclass_for_subclass(self):
+        # Test to ensure that inheritance from a dataclass works correctly
+        # and that the subclass is also recognized as a dataclass.
+        @dataclass
+        class X:
+            y: int
+
+        class Z(X):
+            z: int = 0  # Adding a default field to demonstrate inheritance and dataclass behavior
+
+        # Check if X is a dataclass
+        self.assertTrue(is_dataclass(X), "X should be a dataclass")
+        # Check if Z is a dataclass, it should inherit the dataclass behavior from X
+        self.assertTrue(is_dataclass(Z), "Z should be a dataclass because it inherits from X")
```

Comment:
We should also check that an instance of `Z` returns True from `is_dataclass`

Reply:
Added another assertion to check the same for instances